include/swift/AST/AutoDiff.h

```diff
@@ -531,37 +531,45 @@ inline llvm::raw_ostream &operator<<(llvm::raw_ostream &s,
   return s;
 }
 
-/// The kind of an associated function.
-struct AutoDiffAssociatedFunctionKind {
+/// The kind of an linear map.
+struct AutoDiffLinearMapKind {
   enum innerty : uint8_t {
-     // The Jacobian-vector products function.
-     JVP = 0,
-     // The vector-Jacobian products function.
-     VJP = 1
+    // The differential function.
+    Differential = 0,
+    // The pullback function.
+    Pullback = 1
   } rawValue;
 
-  AutoDiffAssociatedFunctionKind() = default;
-  AutoDiffAssociatedFunctionKind(innerty rawValue) : rawValue(rawValue) {}
-  explicit AutoDiffAssociatedFunctionKind(StringRef string);
+  AutoDiffLinearMapKind() = default;
+  AutoDiffLinearMapKind(innerty rawValue) : rawValue(rawValue) {}
   operator innerty() const { return rawValue; }
 };
 
-/// The kind of an linear map.
-struct AutoDiffLinearMapKind {
+/// The kind of an associated function.
+struct AutoDiffAssociatedFunctionKind {
   enum innerty : uint8_t {
-     // The differential function.
-     Differential = 0,
-     // The pullback function.
-     Pullback = 1
+   // The Jacobian-vector products function.
+   JVP = 0,
+   // The vector-Jacobian products function.
+   VJP = 1
   } rawValue;
 
-  AutoDiffLinearMapKind() = default;
-  AutoDiffLinearMapKind(innerty rawValue) : rawValue(rawValue) {}
+  AutoDiffAssociatedFunctionKind() = default;
+  AutoDiffAssociatedFunctionKind(innerty rawValue) : rawValue(rawValue) {}
+  explicit AutoDiffAssociatedFunctionKind(StringRef string);
   operator innerty() const { return rawValue; }
+  AutoDiffLinearMapKind getLinearMapKind() {
+    switch (*this) {
+    case AutoDiffAssociatedFunctionKind::JVP:
```

Comment:
No need for a switch since they are integers that align with each other directly. Just do a type cast.

Reply:
How should I write the code exactly?

Reply:
```cpp
AutoDiffLinearMapKind getLinearMapKind() {
    return (AutoDiffLinearMapKind::innerty)rawValue;
}
```

Reply:
Thanks! Done in https://github.com/apple/swift/pull/26557/commits/d41d4b8380c340490bdc9ae513b1d9cd1a5a9d30.